test/harvest/ansible/roles/checkout/tasks/main.yml

```diff
@@ -0,0 +1,14 @@
+---
+
+- name: Ensure local working copy does not exist
+  ansible.builtin.file:
+    path: "{{ repo_working_copy }}"
+    state: "absent"
+
+- name: Checkout code
+  ansible.builtin.git:
+    repo: "{{ agent_repo_url }}"
+    dest: "{{ repo_working_copy }}"
+    version: "{{ agent_version }}"
```

Comment:
we will also need to run this for a PR/branch/local, maybe better to use current version of repo? (given that this file is in the repo already)